hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -1955,8 +1981,13 @@ vlapic_rdmsr(struct vcpu *vcpu, uint32_t msr, uint64_t *rval)
 		break;
 
 	default:
-		dev_dbg(ACRN_DBG_LAPIC,
-			"Invalid vlapic msr 0x%x access\n", msr);
+		if (is_x2apic_msr(msr)) {
+			error = vlapic_x2apic_access(vcpu);
```

Comment:
I may lose something. This API is not emulating the rdmsr..  Where is it?

Reply:
read or write is handled in vlapic_x2apic_access